### PR TITLE
Catelog of my things: Ruby Capstone Project

### PR DESCRIPTION
**Pull request by AlaaShaheen2**
Hi!

As part of the Ruby final project, we in a group of two students followed the project requirements and have done the following:

- Created 6 classes that were supposed to be done by two students.
- Preserved data into JSON files.
- Add UNIT tests for all the methods.

- [ ] This is screen shoots for our terminal for the project: 
Screenshot 2022-12-03 at 23 27 52

Screenshot 2022-12-03 at 23 28 02

- [ ] This a link for our recording video 👍 
[The link](https://drive.google.com/file/d/12LAwv3vPFEP_92oHQP5scctHDwTSAYdb/view?usp=sharing)

- [ ] The second video for highlight about a piece of the code you find interesting or are proud of making.
- [ ] [here](https://drive.google.com/file/d/1-uFnf3FQvjFG2h3AulTewJLkgRhmlnO3/view?usp=sharing)

Thank you for taking the time to review our project.

### CODE REVIEW

File: music_album.rb [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+  def initialize(publish_date, on_spotify)
+    super()
+    @id = Random.rand(1..200)
+    @on_spotify = on_spotify
+    @publish_date = Date.parse(publish_date)
+  end
+
+  def can_be_archived?
+    true if super || on_spotify == true
+  end
```
Comment (mikethreels) [inline music_album.rb:16]: - [ ] this method should return true if the parent's method  returns true AND if on_spotify equals true

File: game.rb [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+  def initialize(multiplayer, _last_played, archived: false)
+    super(archived: archived)
+    @multiplayer = multiplayer
+    @last_played_date = Date.parse(last_play)
+  end
+
+  def can_be_archived?
+    check = Time.now.year - @last_play.year > 2
+    archived && check
+  end
```
Comment (mikethreels) [inline game.rb:16]: - [ ] this method should return true if the parent's method  returns true AND if last_played_at is older than 2 years

**Review by mikethreels — CHANGES_REQUESTED** [review]
Hi Team, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- linters are passing

## Required Changes ♻️
- [ ] The video should contain a highlight about a piece of the code you find interesting or are proud of making.
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: music_album.rb (hunk 1 above)
Comment (AlaaShaheen2) [reply music_album.rb]: Done 
thank you

File: game.rb (hunk 2 above)
Comment (AlaaShaheen2) [reply game.rb]: @mikethreels  also done thx 👍

**Comment by AlaaShaheen2** [comment]
@mikethreels  we recorded new video and put it on the pull request

**Review by Bondok6 — APPROVED** [review]
## Hi @AlaaShaheen2 and @mikethreels 🙋‍♂️, 

In my opinion, your project is ready for the presentation!  There is nothing else to say other than... it's time to merge it :shipit: 

- [OPTIONAL] I just suggest customizing your MIT.md file, add your name and customize it 👍




Good luck! 🎉 


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**Pull request merged**